lib/pbench/server/api/resources/__init__.py

```diff
@@ -0,0 +1,686 @@
+
+            Any authenticated ADMIN user can access any private data.
+
+        for API_OPERATION.UPDATE:
+
+            An authenticated user is required.
+
+            Any authenticated user can update their own data.
+
+            Any authenticated ADMIN user can update any data.
+
+        Args:
+            user: The user parameter to the API, or None
+            access: The access parameter to the API, or None
+
+        Raises:
+            UnauthorizedAccess The user isn't authorized for the requested
+                access.
+        """
+        authorized_user: User = Auth.token_auth.current_user()
+        authorized = True
+        self.logger.debug(
+            "Authorizing {} access for {} to user {} with access {}",
+            self.role,
+            authorized_user,
+            user,
+            access,
+        )
+        if self.role != API_OPERATION.READ or access == Dataset.PRIVATE_ACCESS:
+            if authorized_user is None:
+                self.logger.warning(
+                    "Attempt to {} user {} data without login", self.role, user
+                )
+                authorized = False
+            elif user != authorized_user.username and not authorized_user.is_admin():
+                self.logger.warning(
+                    "Unauthorized attempt by {} to {} user {} data",
+                    authorized_user,
+                    self.role,
+                    user,
+                )
+                authorized = False
+        if not authorized:
+            raise UnauthorizedAccess(authorized_user, self.role, user, access)
+
+    def _dispatch(self, method: Callable, request: Request) -> Response:
+        """
+        This is a common front end for HTTP operations, which will validate and
+        normalize the request JSON document if there's a class schema and the
+        operation isn't GET (which doesn't support a request payload).
+
+        Args:
+            method: A reference to the implementation method
+            request: The flask Request object containing payload and headers
+
+        Returns:
+            Flask Response object generally constructed implicitly from a JSON
+            payload and HTTP status.
+        """
+        json_data = request.get_json(silent=True)
+        if self.schema and method != self._get:
+            try:
+                new_data = self.schema.validate(json_data)
+            except UnverifiedUser as e:
+                self.logger.warning("{}", str(e))
+                abort(HTTPStatus.FORBIDDEN, message=str(e))
+            except SchemaError as e:
+                self.logger.warning(
+                    "{}: {} on {!r}", self.__class__.__name__, str(e), json_data
+                )
+                abort(HTTPStatus.BAD_REQUEST, message=str(e))
+            except Exception as e:
+                self.logger.exception("POST unexpected {}: {}", e.__class__.__name__, e)
```

Comment:
The message says 'POST', but this could be any method which accepts a payload, right?

Reply:
Yeah, refactoring.